### PR TITLE
use getsystemvalue-functions in Mailer.php

### LINKED ISSUE #38957: [Bug]: Nextcloud 26.0.2 email notification error
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

hello

there is a bug in mail notification setting.

in log i have:

```
Exception: Symfony\Component\Mailer\Transport\Smtp\Stream\SocketStream::setTimeout(): Argument #1 ($timeout) must be of type float, string given, called in /var/www/nextcloud/lib/private/Mail/Mailer.php on line 305 in file '/var/www/nextcloud/3rdparty/symfony/mailer/Transport/Smtp/Stream/SocketStream.php' line 37
```

### Steps to reproduce

in config.php i try change 
`"mail_smtptimeout"  => '30',`  to `"mail_smtptimeout"  => 30, `
but no efect.

so i delete `"mail_smtptimeout"  => '30'` from config.php
still nothing.

chnge setting from web admin panel restore  `"mail_smtptimeout"  => '30',` to config.php

only delete `"mail_smtptimeout"  => '30'`, from config.php
and
change  
`/var/www/nextcloud/3rdparty/symfony/mailer/Transport/Smtp/Stream/SocketStream.php  in line 47
`
from 
`return $this->timeout ?? (float) \ini_get('default_socket_timeout');`
to:
`return $this->timeout ?? (float) \ini_get(30);`

help me to workaround problem

### Expected behavior

can you fix it?

### Installation method

None

### Nextcloud Server version

26

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

PostgreSQL

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [x] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by szaimen**
Fix https://github.com/nextcloud/server/issues/38957

### CODE REVIEW

File: lib/private/Mail/Mailer.php [hunk 1]
```diff
@@ -292,7 +292,7 @@ protected function getSmtpInstance(): EsmtpTransport {
 		// either null or true - if nothing is passed, let the symfony mailer figure out the configuration by itself
 		$mailSmtpsecure = ($this->config->getSystemValue('mail_smtpsecure', null) === 'ssl') ? true : null;
```
Comment (szaimen) [inline lib/private/Mail/Mailer.php:293]: Maybe?
```suggestion
		$mailSmtpsecure = ($this->config->getSystemValueString('mail_smtpsecure', '') === 'ssl') ? true : null;
```

File: lib/private/Mail/Mailer.php [hunk 2]
```diff
@@ -301,11 +301,11 @@ protected function getSmtpInstance(): EsmtpTransport {
...
+		$stream->setTimeout($this->config->getSystemValueInt('mail_smtptimeout', 10));
 
 		if ($this->config->getSystemValueBool('mail_smtpauth', false)) {
-			$transport->setUsername($this->config->getSystemValue('mail_smtpname', ''));
-			$transport->setPassword($this->config->getSystemValue('mail_smtppassword', ''));
+			$transport->setUsername($this->config->getSystemValueString('mail_smtpname', ''));
+			$transport->setPassword($this->config->getSystemValueString('mail_smtppassword', ''));
 		}
 
 		$streamingOptions = $this->config->getSystemValue('mail_smtpstreamoptions', []);
```
Comment (szaimen) [inline lib/private/Mail/Mailer.php:311]: Maybe?
```suggestion
		$streamingOptions = (array)$this->config->getSystemValue('mail_smtpstreamoptions', []);
```

**Review by kesselb — APPROVED** [review]

**Review by come-nc — APPROVED** [review]

**Pull request merged**